osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+(function() {
+    var FormValidator = function() {
+        this.submitButton = document.getElementById('submit-id-submit');
+        this.extractionFormatCheckboxes = document.querySelectorAll('#div_id_formats input[type="checkbox"]');
+
+
+        this.validity = {
+            'extractionFormats': false
+        };
+
+        this.isFormValid = function() {
+            var isValid = true;
+            Object.keys(this.validity).forEach(function(key){
+                isValid = isValid && this.validity[key];
+            }.bind(this));
+            return isValid;
+        };
+
+        this.setSubmitButtonState = function() {
+            this.submitButton.disabled = !this.isFormValid();
+        }
+
+        this.validateExtractionFormatCheckboxes = function() {
+            this.validity['extractionFormats'] = (document.querySelectorAll('#div_id_formats input[type="checkbox"]:checked').length > 0);
+            this.setSubmitButtonState();
+            Array.prototype.forEach.call(this.extractionFormatCheckboxes, function(checkbox) {
+                if(this.validity['extractionFormats']) {
+                    checkbox.setCustomValidity('');
+                } else {
+                    checkbox.setCustomValidity('Please choose at least one export format!');
```

Comment:
Is this string user-visible? If so, shouldn't it come from a template and be translated?